Cesium3DTilesSelection/src/EllipsoidTilesetLoader.cpp

```diff
@@ -0,0 +1,279 @@
+#include <CesiumGeometry/QuadtreeTileID.h>
+#include <CesiumGeometry/QuadtreeTilingScheme.h>
+#include <CesiumGeospatial/BoundingRegion.h>
+#include <CesiumGeospatial/Cartographic.h>
+#include <CesiumGeospatial/Ellipsoid.h>
+#include <CesiumGeospatial/GeographicProjection.h>
+#include <CesiumGeospatial/GlobeRectangle.h>
+#include <CesiumGeospatial/calcQuadtreeMaxGeometricError.h>
+#include <CesiumGltf/Accessor.h>
+#include <CesiumGltf/AccessorSpec.h>
+#include <CesiumGltf/Buffer.h>
+#include <CesiumGltf/BufferCesium.h>
+#include <CesiumGltf/BufferView.h>
+#include <CesiumGltf/Mesh.h>
+#include <CesiumGltf/MeshPrimitive.h>
+#include <CesiumGltf/Model.h>
+#include <CesiumGltf/Node.h>
+#include <CesiumGltf/Scene.h>
+
+#include <glm/detail/type_vec3.hpp>
+#include <glm/ext/matrix_transform.hpp>
+#include <glm/fwd.hpp>
+
+#include <cstddef>
+#include <cstdint>
+#include <cstring>
+#include <memory>
+#include <optional>
+#include <type_traits>
+#include <variant>
+#include <vector>
+
+using namespace Cesium3DTilesContent;
+
+namespace Cesium3DTilesSelection {
+EllipsoidTilesetLoader::EllipsoidTilesetLoader(const Ellipsoid& ellipsoid)
+    : _projection(ellipsoid),
+      _tilingScheme(
+          _projection.project(_projection.MAXIMUM_GLOBE_RECTANGLE),
+          2,
+          1) {}
+
+/*static*/ std::unique_ptr<Tileset> EllipsoidTilesetLoader::createTileset(
+    const TilesetExternals& externals,
+    const TilesetOptions& options) {
+  std::unique_ptr<EllipsoidTilesetLoader> pCustomLoader =
+      std::make_unique<EllipsoidTilesetLoader>(options.ellipsoid);
+  std::unique_ptr<Tile> pRootTile =
+      std::make_unique<Tile>(pCustomLoader.get(), TileEmptyContent{});
+
+  pRootTile->setRefine(TileRefine::Replace);
+  pRootTile->setUnconditionallyRefine();
+
+  std::vector<Tile> children;
+  uint32_t rootTilesX = pCustomLoader->_tilingScheme.getRootTilesX();
+  children.reserve(rootTilesX);
+
+  for (uint32_t x = 0; x < rootTilesX; x++) {
+    pCustomLoader->createChildTile(
+        *pRootTile,
+        children,
+        QuadtreeTileID{0, x, 0});
+  }
+
+  pRootTile->createChildTiles(std::move(children));
+
+  return std::make_unique<Tileset>(
+      externals,
+      std::move(pCustomLoader),
+      std::move(pRootTile),
+      options);
+}
+
+Future<TileLoadResult>
+EllipsoidTilesetLoader::loadTileContent(const TileLoadInput& input) {
+  return input.asyncSystem.createResolvedFuture(TileLoadResult{
+      createModel(createGeometry(input.tile)),
+      Axis::Z,
+      std::nullopt,
+      std::nullopt,
+      std::nullopt,
+      nullptr,
+      {},
+      TileLoadResultState::Success});
+}
+
+TileChildrenResult EllipsoidTilesetLoader::createTileChildren(
+    const Tile& tile,
+    const CesiumGeospatial::Ellipsoid& ellipsoid) {
+  // We don't use the ellipsoid parameter, since we already have an ellipsoid
+  // specified, but if we don't use the variable we'll get an error. This line
+  // counts as using it.
+  (void)ellipsoid;
```

Comment:
The more common pattern here is to declare the parameter as either:

```cpp
const CesiumGeospatial::Ellipsoid& /*ellipsoid*/
```

If the parameter has no name, the compiler won't flag it as unused. Or, you do this:

```cpp
[[maybe_unused]] const CesiumGeospatial::Ellipsoid& ellipsoid
```

This is the better option when the parameter is sometimes used. For example, when it's only used in an assertion which gets removed in release builds.